src/Service/Program.cs

```diff
@@ -62,46 +64,34 @@ public static bool StartEngine(string[] args)
                 });
 
         /// <summary>
-        /// Iterate through args and based on values present
-        /// set the appropriate log level. If --LogLevel is present
-        /// the next value in args must be "0" through "6", or
-        /// --LogLevel must be the last element in args. In any other
-        /// case we throw an exception. If --LogLevel is the last element
-        /// in args then we ignore it to maintain engine's behavior prior
-        /// to this change.
+        /// Using System.CommandLine Parser to parse args and return
+        /// the correct log level. We save if there is a log level in args through
+        /// the out param. For log level out of range we throw an exception.
         /// </summary>
         /// <param name="args">array that may contain log level information.</param>
         /// <param name="isLogLevelOverridenByCli">sets if log level is found in the args.</param>
         /// <returns>Appropriate log level.</returns>
         private static LogLevel GetLogLevelFromCommandLineArgs(string[] args, out bool isLogLevelOverridenByCli)
         {
-            for (int i = 0; i < args.Length; i++)
-            {
-                if (args[i].Equals("--LogLevel"))
-                {
-                    if (args.Length <= i + 1)
-                    {
-                        break;
-                    }
+            Command cmd = new(name: "start");
```

Comment:
can we also update line 124 in the same file ( `DisableHttpsRedirectionIfNeeded()` ). where we are running the loop as well.